Zend/zend_partial.c

```diff
@@ -0,0 +1,645 @@
+/*
+   +----------------------------------------------------------------------+
+   | Zend Engine                                                          |
+   +----------------------------------------------------------------------+
+   | Copyright (c) Zend Technologies Ltd. (http://www.zend.com)           |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+   | Authors: krakjoe <krakjoe@php.net>                                   |
+   +----------------------------------------------------------------------+
+*/
+#include "zend.h"
+#include "zend_API.h"
+#include "zend_closures.h"
+#include "zend_interfaces.h"
+#include "zend_partial.h"
+#include "zend_partial_arginfo.h"
+
+zend_class_entry *zend_ce_partial;
+zend_object_handlers zend_partial_handlers;
+
+#define ZEND_PARTIAL_PROPERTY_ERROR() \
+	zend_throw_error(NULL, "Partials cannot have properties")
+	
+#define ZEND_PARTIAL_METHOD_ERROR() \
+	zend_throw_error(NULL, "Partials do not implement this method")
+
+typedef struct _zend_partial_applied {
+    zval           This;
+    uint32_t       argc;
+    zval          *argv;
+    zend_array    *named;
+} zend_partial_applied;
+
+typedef struct _zend_partial {
+    zend_object            std;
+    zend_function          prototype;
```

Comment:
The prototype is what we give to reflection, it has the correct arginfo and is marked abstract so is not executable